### PR TITLE
Add SIIM ACR Pneumothorax dataset

### PR DESCRIPTION
**Pull request by anwai98**
This is a dataset for segmenting pneumothorax in chest x-rays.

### CODE REVIEW

**Comment by anwai98** [comment]
In addition, this PR also adds the feature to download files from kaggle using the Kaggle API.

**Comment by anwai98** [comment]
Hi @constantinpape 

This is a nice spot to review a few new features:
- Downloads are now possible from Kaggle (some simple steps in the beginning to make the API work)
- In `ImageCollectionDataset`, it allows now to accept `patch_shape` as None, and load the entire image as it is.
- Added a new transformation function called `ResizeInputs`, which is the `skimage.transform.resize` operation to resize the inputs based on a desired patch shape.

Let me know if you spot something.

File: torch_em/data/image_collection_dataset.py [hunk 1]
```diff
@@ -95,11 +96,16 @@ def ndim(self):
         return self._ndim
 
     def _sample_bounding_box(self, shape):
+        if self.patch_shape is None:
+            patch_shape_for_bb = shape
+        else:
+            patch_shape_for_bb = self.patch_shape
```
Comment (constantinpape) [inline torch_em/data/image_collection_dataset.py:102]: I would also include the `bb_start` logic in this `if-else` and set it to `[0]* len(shape)` if the patch shape is None. 
While I think the current logic woks, it is not very readable.

**Review by constantinpape — COMMENTED** [review]
- Some minor things can be improved in the handling if `None` patch shape, but looks good overall.
- This logic should also be implemented for the `SegmentationDataset`. Otherwise there will be a disconnect in the functionality.
- Some changes to the resize transformation.

File: torch_em/transform/generic.py [hunk 2]
```diff
@@ -72,6 +72,28 @@ def __call__(self, *inputs):
         return outputs
 
 
+class ResizeInputs:
+    def __init__(self, target_shape, is_label=False):
+        self.target_shape = target_shape
+        self.is_label = is_label
+
+    def __call__(self, inputs):
+        if self.is_label:
```
[the remark this answers is not in the document]
Comment (anwai98) [reply torch_em/transform/generic.py]: Thanks for spotting. Taken care of in a recent commit.

File: torch_em/data/image_collection_dataset.py (hunk 1 above)
Comment (anwai98) [reply torch_em/data/image_collection_dataset.py:103]: Thanks!

**Comment by anwai98** [comment]
Thanks for the mentions!

> Re: This logic should also be implemented for the `SegmentationDataset`. Otherwise there will be a disconnect in the functionality.

I was thinking of doing it in another relevant PR once this is merged, where segmentation dataset is used (ideally for a 3d case like ACDC). Would that work, or should I take care of this in this PR?

**Comment by constantinpape** [comment]
> Would that work, or should I take care of this in this PR?

Yes, that works!

**Review by constantinpape — APPROVED** [review]

**Pull request merged**